flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/expressions/RexNodeConverter.java

```diff
@@ -79,25 +112,196 @@
+
+		// crypto hash
+		SIMPLE_DEF_SQL_OPERATOR_MAPPING.put(BuiltInFunctionDefinitions.MD5, FlinkSqlOperatorTable.MD5);
+		SIMPLE_DEF_SQL_OPERATOR_MAPPING.put(BuiltInFunctionDefinitions.SHA2, FlinkSqlOperatorTable.SHA2);
+		SIMPLE_DEF_SQL_OPERATOR_MAPPING.put(BuiltInFunctionDefinitions.SHA224, FlinkSqlOperatorTable.SHA224);
+		SIMPLE_DEF_SQL_OPERATOR_MAPPING.put(BuiltInFunctionDefinitions.SHA256, FlinkSqlOperatorTable.SHA256);
+		SIMPLE_DEF_SQL_OPERATOR_MAPPING.put(BuiltInFunctionDefinitions.SHA384, FlinkSqlOperatorTable.SHA384);
+		SIMPLE_DEF_SQL_OPERATOR_MAPPING.put(BuiltInFunctionDefinitions.SHA512, FlinkSqlOperatorTable.SHA512);
+		SIMPLE_DEF_SQL_OPERATOR_MAPPING.put(BuiltInFunctionDefinitions.SHA1, FlinkSqlOperatorTable.SHA1);
+
+		// etc
+		// SIMPLE_DEF_SQL_OPERATOR_MAPPING.put(InternalFunctionDefinitions.THROW_EXCEPTION, FlinkSqlOperatorTable.THROW_EXCEPTION);
+
+	}
+
 	public RexNodeConverter(RelBuilder relBuilder) {
 		this.relBuilder = relBuilder;
 		this.typeFactory = (FlinkTypeFactory) relBuilder.getRexBuilder().getTypeFactory();
 	}
 
+	// TODO removed later after PlanExpression is merged
 	public RexNode visit(UnresolvedCallExpression call) {
-		switch (call.getFunctionDefinition().getKind()) {
+		FunctionDefinition func = call.getFunctionDefinition();
+		switch (func.getKind()) {
 			case SCALAR:
-				return translateScalarCall(call.getFunctionDefinition(), call.getChildren());
-			default: throw new UnsupportedOperationException();
+				if (func instanceof ScalarFunctionDefinition) {
+					ScalarFunction scalaFunc = ((ScalarFunctionDefinition) func).getScalarFunction();
+					List<RexNode> child = convertCallChildren(call.getChildren());
+					SqlFunction sqlFunction = UserDefinedFunctionUtils.createScalarSqlFunction(
+							scalaFunc.functionIdentifier(),
+							scalaFunc.toString(),
+							scalaFunc,
+							typeFactory);
+					return relBuilder.call(sqlFunction, child);
+				} else {
+					return visitBuiltInFunc(call.getFunctionDefinition(), call.getChildren());
+				}
+
+			default:
+				throw new UnsupportedOperationException();
 		}
 	}
 
 	@Override
 	public RexNode visit(CallExpression call) {
-		switch (call.getFunctionDefinition().getKind()) {
-			case SCALAR:
-				return translateScalarCall(call.getFunctionDefinition(), call.getChildren());
-			default: throw new UnsupportedOperationException();
+		FunctionDefinition func = call.getFunctionDefinition();
+		if (func instanceof ScalarFunctionDefinition) {
+			ScalarFunction scalaFunc = ((ScalarFunctionDefinition) func).getScalarFunction();
+			List<RexNode> child = convertCallChildren(call.getChildren());
+			SqlFunction sqlFunction = UserDefinedFunctionUtils.createScalarSqlFunction(
+					scalaFunc.functionIdentifier(),
+					scalaFunc.toString(),
+					scalaFunc,
+					typeFactory);
+			return relBuilder.call(sqlFunction, child);
+		} else if (func instanceof TableFunctionDefinition) {
+			throw new UnsupportedOperationException(func.toString());
```

Comment:
Why not?

Reply:
There is no possible rich here.

Reply:
Will 	throw an exception which explicitly announce that "There is no possible rich here."